test/tests/functional/pbs_cgroups_hook.py

```diff
@@ -457,76 +451,103 @@ def load_hook(self, filename):
+                                       overwrite=True)
+        # Add the configuration
+        self.load_config(self.cfg0)
 
     def load_config(self, cfg):
         """
         Create a hook configuration file with the provided contents.
         """
-        cfg_file = os.path.join(os.sep, 'tmp', self.random_name())
-        with open(cfg_file, "w") as fd:
-            fd.write(cfg)
+        (fd, fn) = self.du.mkstemp()
+        os.write(fd, cfg)
+        os.close(fd)
         a = {'content-type': 'application/x-config',
              'content-encoding': 'default',
-             'input-file': cfg_file}
+             'input-file': fn}
         self.server.manager(MGR_CMD_IMPORT, HOOK, a, self.hook_name)
-        os.remove(cfg_file)
+        os.remove(fn)
+        self.mom.log_match('pbs_cgroups.CF;copy hook-related ' +
+                           'file request received',
+                           max_attempts=3,
+                           starttime=self.server.ctime)
 
     def set_vntype(self, typestring='myvntype'):
         """
         Set the vnode type for the local mom.
         """
         pbs_home = self.server.pbs_conf['PBS_HOME']
         vntype_file = os.path.join(pbs_home, 'mom_priv', 'vntype')
-        with open(vntype_file, "w") as fd:
-            fd.write(typestring)
+        self.logger.info("Setting vntype to %s in %s" %
+                         (typestring, vntype_file))
+        (fd, fn) = self.du.mkstemp()
+        os.write(fd, typestring)
+        os.close(fd)
+        ret = self.du.run_copy(hosts=self.mom.hostname, src=fn,
+                               dest=vntype_file, sudo=True, uid='root',
+                               gid='root', mode=0644)
+        os.remove(fn)
+        if ret['rc'] != 0:
+            self.skipTest("pbs_cgroups_hook: need root privileges")
 
     def remove_vntype(self):
         """
         Unset the vnode type for the local mom.
         """
         pbs_home = self.server.pbs_conf['PBS_HOME']
         vntype_file = os.path.join(pbs_home, 'mom_priv', 'vntype')
-        if os.path.isfile(vntype_file):
-            os.remove(vntype_file)
+        ret = self.du.rm(hostname=self.mom.hostname, path=vntype_file,
+                         force=True, sudo=True)
+        if not ret:
+            self.skipTest("pbs_cgroups_hook: need root privileges")
 
-    # Wait for up to ten seconds for a file to appear. Remove it if it does.
-    def wait_and_remove(self, filename):
+    def wait_and_remove_file(self, filename):
         """
         Wait up to ten seconds for a file to appear and then remove it.
         """
+        self.logger.info("Removing file: %s" % filename)
+        if not filename:
+            return str()
         for _ in range(10):
             if not os.path.isfile(filename):
                 break
             try:
-                os.remove(filename)
+                self.du.rm(hostname=self.mom.hostname, path=filename,
+                           force=True, sudo=True)
```

Comment:
Also add runas=TEST_USER

Reply:
Making this change would cause the functions that modify vntype to fail.

Reply:
Added.